File: java_apx/src/main/java/net/sympower/citizen/apx/scheduler/ProviderCallScheduler.java
package net.sympower.citizen.apx.scheduler;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.scheduling.annotation.Scheduled;

public class ProviderCallScheduler {

    private static final Logger LOGGER = LoggerFactory.getLogger(ProviderCallScheduler.class);

    @Scheduled(cron = "${cron.every-hour}")
    public void callProvider() {
        LOGGER.info("Execute Provider caller: ${LocalDateTime.now()}");
        LOGGER.warn("I'm doing nothing with data like this: ${dataResponseService.getDataResponse()}");
    }
}

File: src/main/kotlin/net/sympower/citizen/apx/scheduler/ProviderCallScheduler.kt
package net.sympower.citizen.apx.scheduler

import net.sympower.citizen.apx.service.DataResponseService
import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Scheduled
import java.time.LocalDateTime

class ProviderCallScheduler(
    private val dataResponseService: DataResponseService
) {

    companion object {
        private val LOGGER = LoggerFactory.getLogger(ProviderCallScheduler::class.java)
    }

    @Scheduled(cron = "\${cron.every-hour}")
    fun callProvider() {
        LOGGER.info("Execute Provider caller: ${LocalDateTime.now()}")
        LOGGER.warn("I'm doing nothing with data like this: ${dataResponseService.getDataResponse()}")
    }

}